vaadin-map-flow-parent/vaadin-map-flow/src/test/java/com/vaadin/flow/component/map/configuration/layer/FeatureLayerTest.java

```diff
@@ -58,6 +66,130 @@ public void removeFeature() {
                 .propertyChange(Mockito.any());
     }
 
+    @Test
+    public void enableClustering() {
+        MarkerFeature feature1 = new MarkerFeature();
+        MarkerFeature feature2 = new MarkerFeature();
+        featureLayer.addFeature(feature1);
+        featureLayer.addFeature(feature2);
+
+        featureLayer.setClusteringEnabled(true);
+
+        Assert.assertTrue(featureLayer.isClusteringEnabled());
+        Assert.assertTrue(featureLayer.getSource() instanceof ClusterSource);
+        Assert.assertEquals(2, featureLayer.getFeatures().size());
+        Assert.assertTrue(featureLayer.getFeatures().contains(feature1));
+        Assert.assertTrue(featureLayer.getFeatures().contains(feature2));
+    }
+
+    @Test
+    public void enableClustering_copiesSourceProperties() {
+        VectorSource.Options options = new VectorSource.Options();
+        options.setProjection("EPSG:4326");
+        options.setAttributionsCollapsible(false);
+        options.setAttributions(List.of("Test Attribution"));
+        VectorSource customSource = new VectorSource(options);
+        featureLayer.setSource(customSource);
+
+        featureLayer.setClusteringEnabled(true);
+
+        ClusterSource clusterSource = (ClusterSource) featureLayer.getSource();
+        Assert.assertEquals(List.of("Test Attribution"),
+                clusterSource.getAttributions());
+        Assert.assertEquals(customSource.isAttributionsCollapsible(),
+                clusterSource.isAttributionsCollapsible());
+        Assert.assertEquals(customSource.getProjection(),
+                clusterSource.getProjection());
+    }
+
+    @Test
+    public void disableClustering() {
+        MarkerFeature feature1 = new MarkerFeature();
+        MarkerFeature feature2 = new MarkerFeature();
+        featureLayer.setClusteringEnabled(true);
+        featureLayer.addFeature(feature1);
+        featureLayer.addFeature(feature2);
+
+        featureLayer.setClusteringEnabled(false);
+
+        Assert.assertFalse(featureLayer.isClusteringEnabled());
+        Assert.assertFalse(featureLayer.getSource() instanceof ClusterSource);
+        Assert.assertEquals(2, featureLayer.getFeatures().size());
+        Assert.assertTrue(featureLayer.getFeatures().contains(feature1));
+        Assert.assertTrue(featureLayer.getFeatures().contains(feature2));
+    }
+
+    @Test
+    public void setClusterDistance() {
+        featureLayer.setClusterDistance(50);
+        Assert.assertEquals(50, featureLayer.getClusterDistance());
+
+        featureLayer.setClusteringEnabled(true);
+        ClusterSource clusterSource = (ClusterSource) featureLayer.getSource();
+        Assert.assertEquals(50, clusterSource.getDistance());
+
+        featureLayer.setClusterDistance(100);
+        Assert.assertEquals(100, clusterSource.getDistance());
+    }
```

Comment:
It makes more sense to set it to a different value from the default one, IMO:
```suggestion
    public void setClusterDistance() {
        featureLayer.setClusterDistance(10);
        Assert.assertEquals(10, featureLayer.getClusterDistance());

        featureLayer.setClusteringEnabled(true);
        ClusterSource clusterSource = (ClusterSource) featureLayer.getSource();
        Assert.assertEquals(10, clusterSource.getDistance());

        featureLayer.setClusterDistance(100);
        Assert.assertEquals(100, clusterSource.getDistance());
    }
```